File: src/main/java/com/numbers/discordbot/commands/ProgrammerHumorCommand.java
package com.numbers.discordbot.commands;

import com.numbers.discordbot.*;
import com.numbers.discordbot.filter.*;
import com.numbers.discordbot.network.reddit.*;
import java.awt.*;
import java.util.List;
import java.util.*;
import sx.blah.discord.handle.impl.events.guild.channel.message.*;
import sx.blah.discord.util.*;

@Command
public class ProgrammerHumorCommand {

    @Command
    @MessageFilter(eventType = MentionEvent.class, mentionsBot = true, regex = ".*programmerHumor")
    public void handleMention(MentionEvent event, ProgrammerHumor ph){
        List<ProgrammerHumor.ProgrammerPost> urls = ph.getPosts();
        ProgrammerHumor.ProgrammerPost post = urls.get(new Random().nextInt(urls.size()));
        EmbedBuilder builder = new EmbedBuilder().withColor(Color.BLUE).withImage(post.getUrl()).withTitle(post.getTitle());
        
        
        event.getChannel().sendMessage(builder.build());
    }
    
}

File: src/main/java/com/numbers/discordbot/commands/ProgrammerHumorCommand.kt
package com.numbers.discordbot.commands

import com.numbers.discordbot.Command
import com.numbers.discordbot.commands.util.random
import com.numbers.discordbot.commands.util.withDefaultColor
import com.numbers.discordbot.filter.MessageFilter
import com.numbers.discordbot.network.reddit.ProgrammerHumor
import sx.blah.discord.handle.impl.events.guild.channel.message.MentionEvent
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder

@Command(name =  "/r/ProgrammerHumor")
class ProgrammerHumorCommand {

    @Command
    @MessageFilter(eventType = MentionEvent::class, mentionsBot = true, regex = ".*programmerHumor", readableUsage = "programmerHumor")
    fun handleMention(event: MentionEvent, ph: ProgrammerHumor) {
        handlePrefix(event, ph)
    }

    @Command
    @MessageFilter(eventType = MessageEvent::class, prefixCheck = true, regex = "programmerHumor", readableUsage = "programmerHumor")
    fun handlePrefix(event: MessageEvent, ph: ProgrammerHumor){
        val post = ph.posts.random()
        val builder = EmbedBuilder().withDefaultColor().withImage(post.url).withUrl(post.url).withTitle(post.title)

        event.channel.sendMessage(builder.build())
    }

}